internal/agent/config.go

```diff
@@ -136,3 +145,15 @@ func (c *Config) validate() error {
 
 	return c.Cloud.validate()
 }
+
+func (c *Config) logDeprecations(logger hclog.Logger) {
```

Comment:
We could make this function just return deprecation strings, and do the logging at the command level. This would allow us to unit test these messages with config that triggers them?

Reply:
#protip